File: parsers/test_json.py
#!/usr/bin/env python

import json
import sys

#def f_parse_constant(o):
#    raise BaseException

def parse_file(path):

    with open(path, 'r') as f:
    
        data = f.read()

        try:
            o = json.loads(data)
            #o = json.loads(data, parse_constant=f_parse_constant)
            #print(o)
            #if o == None:
            #    sys.exit(1)

        except Exception as e:
            #print("--1")
            #print(e)
            #print("--2")
            sys.exit(1)
    
    """
    s = o.encode('utf-8')
    with open('/Users/nst/Desktop/p.txt', 'wb') as f:
        f.write(s)
    """
    
if __name__ == "__main__":

    path = sys.argv[1]
    parse_file(path)

    sys.exit(0)

File: tests/JSONTestSuite/parsers/test_json.rb
#!/usr/bin/ruby

require 'json'

f = ARGV[0]

o = nil

begin
    puts(f)
    json = File.read(f)
    o = JSON.parse(json, {:quirks_mode => true})
    p o
    
    if o == nil
        exit 1
    else
        exit 0
    end
rescue JSON::ParserError => e
    puts(e)
    exit 1
end
